cmdlineparserlib/include/cmd_line_parser.h

```diff
@@ -0,0 +1,48 @@
+#include<map>
+#include<unordered_map>
+#include<vector>
+#include<string> 
+#include<iostream>
+#include<algorithm>
+#include <cstdio>
+
+namespace cmd_parser
+{
+	
+	class AddOptions
+	{
+	protected:
+
+		std::vector<std::string> so;//for  short command 
+		std::vector<std::string> lo;//for long command
+		std::vector<std::string> s_desc;//for short description
+		std::vector<std::string> l_desc;//for long description
+		std::vector<std::string> data; //for data type of command 
+	public:
+
+
+		AddOptions();
+		~AddOptions();
+		template <typename T>
+		void add_options(std::string short_option, std::string long_option, std::string short_description, std::string long_descriptionl,T val);
+	};
+	//definition  of add_options()
+	template <class T>
+	void cmd_parser::AddOptions::add_options(std::string short_option, std::string long_option, std::string short_description, std::string long_description, T val)
+	{
+		so.push_back('-'+short_option);
+		lo.push_back("--"+long_option);
+		s_desc.push_back(short_description);
+		l_desc.push_back(long_description);
+		data.push_back(static_cast<std::string>(typeid(val).name()));
+
+	}
+	class ParseCmdLine :public AddOptions
```

Comment:
Please make separate .h file for indivisual classes, don't try to put multiple class in single  '.h' file